src/audio/module_adapter/module_adapter.c

```diff
@@ -1399,6 +1488,13 @@ int module_adapter_bind(struct comp_dev *dev, void *data)
 
 	/* find an empty slot in the source_info array */
 	source_index = find_module_source_index(mod_source_info, NULL);
+
+	/* Disable 1 source 1 sink simple copy mode if multiple source buffers */
+	if (mod->simple_copy && source_index > 0) {
```

Comment:
Note: Earlier comment by @ranj063 "what we have to do here is iterate through the sink and source lists and count them and then set the flag if either one is > 1"

Will work next to fix this, and same for unbind.